android/src/main/java/com/magicleap/magicscript/utils/Extensions.kt

```diff
@@ -96,6 +97,58 @@ fun Vector3.rotatedBy(quaternion: Quaternion): Vector3 {
     return Quaternion.rotateVector(quaternion, this)
 }
 
+/**
+ * com.google.ar.sceneform.math.Matrix
+ */
+
+/**
+ * Converts the matrix to a pure rotation matrix (no scale or translation).
+ * It removes the scale with a high precision using BigDecimal class.
+ *
+ * It should be used before extracting a rotation when the matrix contains
+ * also a scale transformation, else the standard decompose functions
+ * (decomposeRotation or extractRotation) will return a very low precision quaternion.
+ *
+ * @param decomposedScale previously decomposed scale from the matrix
+ *
+ * Based on com.google.ar.sceneform.math.Matrix.java
+ */
```

Comment:
Code looks ok, but it's really strange that you need to switch to BigDecimal (Double?) to calculate/extract anything from the Matrix.

Reply:
Have you found any info about this issue in the Internet?

Reply:
Yes, however not directly related to ArCore. From [this](https://computergraphics.stackexchange.com/questions/4281/is-it-possible-to-turn-a-3d-rotation-matrix-4x4-into-its-component-parts-rota) :

Reply:
I also carefully checked the ARCore's decomposition algorithm and compared with others - it's correct. Yes, I'm also surprised, but in this case precision really matter :-)